commit dfeabf4900060248fea4fa39348ba95896f5d87e
Author: Andrea Iacovitti <aiacovitti@libero.it>
Date:   Mon Nov 4 16:22:45 2013 +0100

    - Do not try to remove positioned objects if they are not in the list.
    - Code simplification.

diff --git a/khtml/rendering/render_block.cpp b/khtml/rendering/render_block.cpp
index f9c76de..c75c730 100644
--- a/khtml/rendering/render_block.cpp
+++ b/khtml/rendering/render_block.cpp
@@ -2610,18 +2610,8 @@ void RenderBlock::addOverHangingFloats( RenderBlock *flow, int xoff, int offset,
                 }
             }
 
-            bool already = false;
             // don't insert it twice!
-            if (m_floatingObjects) {
-                QListIterator<FloatingObject*> it2(*m_floatingObjects);
-                while ( it2.hasNext() ) {
-                    if (it2.next()->node == r->node) {
-                        already = true;
-                        break;
-                    }
-                }
-            }
-            if ( !already ) {
+            if (!containsFloat(r->node)) {
                 FloatingObject *floatingObj = new FloatingObject(KDE_CAST_BF_ENUM(FloatingObject::Type, r->type));
                 floatingObj->startY = r->startY - offset;
                 floatingObj->endY = r->endY - offset;
diff --git a/khtml/rendering/render_layer.cpp b/khtml/rendering/render_layer.cpp
index 284933b..5060ff4 100644
--- a/khtml/rendering/render_layer.cpp
+++ b/khtml/rendering/render_layer.cpp
@@ -427,7 +427,7 @@ void RenderLayer::updateWidgetMasks(RenderLayer* rootLayer)
             zx = sc->zIndex();
         }
         if (!needUpdate) {
-            needUpdate = needUpdate || !m_region.isEmpty();
+            needUpdate = !m_region.isEmpty();
             m_region = QRegion();
         }
         if (needUpdate)
diff --git a/khtml/rendering/render_object.cpp b/khtml/rendering/render_object.cpp
index 7185469..3e89062 100644
--- a/khtml/rendering/render_object.cpp
+++ b/khtml/rendering/render_object.cpp
@@ -2389,7 +2389,7 @@ void RenderObject::removeFromObjectLists()
 
     }
 
-    if (isPositioned()) {
+    if (inPosObjectList()) {
         RenderObject *p;
         for (p = parent(); p; p = p->parent()) {
             if (p->isRenderBlock())
